### PR TITLE
Add a method to identify failures happening consistently on a given configuration

### PR DESCRIPTION
**Pull request by marco-c**
Prerequisite of #607

### CODE REVIEW

File: mozci/task.py [hunk 1]
```diff
@@ -451,6 +452,24 @@ def failing_tasks(self):
...
+                if result.group == self.name:
+                    config_to_results[task.configuration].append(result.ok)
+
+        # If there is no config for which we have at least two runs, return None (that is, unknown).
+        if all(len(results) <= 1 for results in config_to_results.values()):
+            return None
+
+        # Return True if there is at least one configuration for which we have only failures, False otherwise.
+        return any(
+            len(results) >= 2 and all(not result for result in results)
```
Comment (ahal) [inline mozci/task.py:469]: nit: I think the `all(not result for result in results)` could be simplified to `not any(results)`?

Comment (ahal) [inline mozci/task.py:469]: The `2` could pretty easily mis-classify a frequent intermittent. Though I know it's a tradeoff as having 3+ retriggers is fairly rare.

File: mozci/task.py [hunk 2]
```diff
@@ -451,6 +452,24 @@ def failing_tasks(self):
             )
         ]
 
+    @memoized_property
+    def is_config_consistent_failure(self) -> Optional[bool]:
+        config_to_results = collections.defaultdict(list)
+        for task in self.tasks:
+            for result in task.results:
+                if result.group == self.name:
+                    config_to_results[task.configuration].append(result.ok)
```
Comment (ahal) [inline mozci/task.py:461]: Unrelated, but this reminded me we still haven't made use of the `test-setting` dict here.

**Review by ahal — APPROVED** [review]
Lgtm!

File: mozci/task.py (hunk 2 above)
Comment (marco-c) [reply mozci/task.py:461]: Indeed, #557. We should do it soon.

**Pull request merged**